pkg/gatewayserver/gatewayserver.go

```diff
@@ -917,7 +917,7 @@ func (gs *GatewayServer) updateConnStats(ctx context.Context, conn connectionEnt
 				case <-time.After(duration):
 				}
 			}
-		case <-refreshTTLTimer.C:
+		case <-updateStatsTimer.C:
```

Comment:
I think we should reset the ticker after exiting the `select` here, in order to avoid having two stats updates in rapid succession (`conn.StatsChanged()` triggers at second 59, then 1 second later `updateStatsTimer.C` triggers again).

Reply:
Yeah good point. I reworked it a bit. We were waiting on the debounce time even if the last time was long ago, which isn't really necessary.

The new code should update anywhere between `UpdateConnectionStatsInterval` and `UpdateConnectionStatsDebounceTime`.